metarecord/admin/classification.py

```diff
@@ -1,10 +1,47 @@
 from django.contrib import admin
+from django.http import HttpResponseForbidden, HttpResponseNotAllowed, HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+from django.urls import path, reverse
 
 from metarecord.models.classification import Classification
 
 
 @admin.register(Classification)
 class ClassificationAdmin(admin.ModelAdmin):
+    change_form_template = "admin/metarecord/classification/change_form.html"
     list_display = ('code', 'version', 'state', 'title', 'function_allowed')
     search_fields = ('code', 'title')
     ordering = ('code', 'version')
+
+    def get_urls(self):
+        urls = super().get_urls()
+        custom_urls = [
+            path(
+                '<path:object_id>/new-version/',
+                self.admin_site.admin_view(self.create_new_version_view),
+                name='metarecord_classification_create-new-version'
+            ),
+        ]
+        return custom_urls + urls
+
+    def create_new_version_view(self, request, object_id):
+        """
+        Create new classification draft version with the information
+        carried over from the latest version
+        """
+        if request.method != 'POST':
+            raise HttpResponseNotAllowed
+
+        if not request.user.has_perm(Classification.CAN_EDIT):
+            raise HttpResponseForbidden
```

Comment:
Hmm, can you actually raise `HttpResponseNotAllowed` and `HttpResponseForbidden`, I don't think they are exceptions?

Reply:
Yeah, you're correct. Good catch 👍